src/main/java/org/candlepin/subscriptions/tally/facts/normalizer/RhsmFactNormalizer.java

```diff
@@ -72,4 +91,31 @@ private boolean isRhel(Map<String, Object> facts) {
             .isEmpty();
     }
 
+    /**
+     * A host is considered unregistered if the last time it was synced passes the configured number
+     * of days. The cutoff will be the at the end of the day for the configured threshold. For example,
+     * if the threshold is 1 day, and the last sync was mid-day yesterday, then the host would be
+     * considered unregistered at midnight tomorrow. This allows the entire day of the threshold day to
+     * wait for a checkin.
+     *
+     * NOTE: If the passed lastSync date is null, it is considered to be registered.
+     *
+     * @param lastSync the last known time that a host sync occured from pinhead to conduit.
+     * @return true if the host is considered unregistered, false otherwise.
+     */
+    private boolean hostUnregistered(OffsetDateTime lastSync) {
+        // If last sync is not present, consider it a valid host.
+        if (lastSync == null) {
+            return false;
+        }
+        return !clock.instant().isBefore(getLastSyncThreshold(lastSync));
+    }
+
+    private Instant getLastSyncThreshold(OffsetDateTime lastSync) {
```

Comment:
This doesn't need all of the type conversions:

```java
OffsetDateTime threshold = lastSync.plusDays(hostSyncThresholdDays);
return LocalTime.MIDNIGHT.adjustInto(threshold.plusDays(1);
```
but if you accept my suggestions, this becomes much simpler:
```java
return OffsetDateTime.now(clock).minusDays(hostSyncThresholdDays);
```

Reply:
DONE